src/hypofuzz/dashboard.py

```diff
@@ -307,6 +306,16 @@ def download_patch(name: str) -> flask.Response:
     )
 
 
+# api design is in flux and will likely evolve alongside our testing needs.
+@app.route("/api/state")  # type: ignore
+def api_state() -> flask.Response:
+    # todo: poll on a fixed interval (even if the page isn't loaded) instead of
+    # on every api request.
+    poll_database()
```

Comment:
relatively easy way to do this: have `poll_database()` record when it last polled, and be a noop if that's within a few seconds - a kinda request-coalescing-flavored TTL setup.

(longer term, we'll want to extend our database interface to support incremental queries, but that's a whole nother design question)